src/dk/aau/cs/ds306e18/tournament/rlbot/configuration/MatchConfig.java

```diff
@@ -0,0 +1,356 @@
+    public final static String GAME_MODE = "game_mode";
+    public final static String GAME_MAP = "game_map";
+    public final static String SKIP_REPLAYS = "skip_replays";
+    public final static String INSTANT_START = "start_without_countdown";
+
+    public final static String PARTICIPANTS_CONFIGURATION_HEADER = "Participant Configuration";
+    public final static String PARTICIPANT_CONFIG_INDEXED = "participant_config_";
+    public final static String PARTICIPANT_TEAM_INDEXD = "participant_team_";
+    public final static String PARTICIPANT_TYPE_INDEXD = "participant_type_";
+    public final static String PARTICIPANT_SKILL_INDEXD = "participant_bot_skill_";
+
+    public final static String MUTATOR_CONFIGURATION_HEADER = "Mutator Configuration";
+    public final static String MUTATOR_MATCH_LENGTH = "Match Length";
+    public final static String MUTATOR_MAX_SCORE = "Max Score";
+    public final static String MUTATOR_OVERTIME = "Overtime";
+    public final static String MUTATOR_GAME_SPEED = "Game Speed";
+    public final static String MUTATOR_BALL_MAX_SPEED = "Ball Max Speed";
+    public final static String MUTATOR_BALL_TYPE = "Ball Type";
+    public final static String MUTATOR_BALL_WEIGHT = "Ball Weight";
+    public final static String MUTATOR_BALL_SIZE = "Ball Size";
+    public final static String MUTATOR_BALL_BOUNCINESS = "Ball Bounciness";
+    public final static String MUTATOR_BOOST_AMOUNT = "Boost Amount";
+    public final static String MUTATOR_RUMBLE = "Rumble";
+    public final static String MUTATOR_BOOST_STRENGTH = "Boost Strength";
+    public final static String MUTATOR_GRAVITY = "Gravity";
+    public final static String MUTATOR_DEMOLISH = "Demolish";
+    public final static String MUTATOR_RESPAWN_TIME = "Respawn Time";
+
+    private File configFile;
+    private GameMap gameMap = GameMap.CHAMPIONS_FIELD;
+    private GameMode gameMode = GameMode.SOCCER;
+    private boolean skipReplays = false;
+    private boolean instantStart = false;
+    private final List<ParticipantInfo> participants = new ArrayList<>();
+
+    private MatchLength matchLength = MatchLength.FIVE_MINUTES;
+    private MaxScore maxScore = MaxScore.UNLIMITED;
+    private Overtime overtime = Overtime.UNLIMITED;
+    private GameSpeed gameSpeed = GameSpeed.DEFAULT;
+    private BallMaxSpeed ballMaxSpeed = BallMaxSpeed.DEFAULT;
+    private BallType ballType = BallType.DEFAULT;
+    private BallWeight ballWeight = BallWeight.DEFAULT;
+    private BallSize ballSize = BallSize.DEFAULT;
+    private BallBounciness ballBounciness = BallBounciness.DEFAULT;
+    private BoostAmount boostAmount = BoostAmount.DEFAULT;
+    private BoostStrength boostStrength = BoostStrength.TIMES_ONE;
+    private RumblePowers rumblePowers = RumblePowers.NONE;
+    private Gravity gravity = Gravity.DEFAULT;
+    private Demolish demolish = Demolish.DEFAULT;
+    private RespawnTime respawnTime = RespawnTime.THREE_SECONDS;
+
+    public MatchConfig() {
```

Comment:
Should an empty `MatchConfig` be allowed or is this an error?

Reply:
Yes. This is required so we can create a default match config when a new tournament is created.